openedx/frontend-component-footer
# feat!: footer legal links

## Related Issues
Product Review Issue: https://github.com/openedx/platform-roadmap/issues/324
https://github.com/mitodl/mitxonline/issues/223

## Description
Legacy LMS provides a way to configure the footer links but the MFE footer has no support for legal links. This PR adds support to configure custom legal links in the MFE footer.

## Testing

- This PR can be tested with the Accounts MFE https://github.com/openedx/frontend-app-account.
- Test without adding the links. The footer should be the same as before.
- Add config links in .env.development:
  - ACCESSIBILITY_URL
  - ABOUT_US_URL
  - HONOR_CODE_URL
  - CONTACT_URL
  - SUPPORT_CENTER_URL
  - SUPPORT_CENTER_TEXT
  - TRADEMARK_TEXT
  - FOOTER_LOGO_ALT_TEXT
  - SHOW_FOOTER_LOGO
  - TERMS_OF_SERVICE_URL
  - PRIVACY_POLICY_URL
- Verify the footer links in the MFE
- Verify the Accessibility URL in the Studio MFE

## Screenshots
1. MFE Footer after our changes
<img width="1742" alt="Screenshot 2024-02-22 at 4 32 31 PM" src="https://github.com/openedx/frontend-component-footer/assets/52656433/fc9662e2-32ac-4445-aa62-a3b3077b9c5d">

2. MFE Footer before these changes
<img width="1738" alt="300859871-205c0ce3-8c0a-4c74-b6e9-dde2233af4b3" src="https://github.com/openedx/frontend-component-footer/assets/52656433/dccdb08c-20da-49dc-87f7-39a87df5c650">

3. Legacy LMS Footer:
<img width="1738" alt="300857051-ee5005ec-4a6c-40c6-ba10-18ca61005b79" src="https://github.com/openedx/frontend-component-footer/assets/52656433/0110c209-ac83-4535-b5f2-1146cf74e78b">

4. Studio MFE footer
<img width="1738" alt="300857498-cf882470-bd2a-4997-9202-ecb7cd885ebb" src="https://github.com/openedx/frontend-component-footer/assets/52656433/0284d612-7e7e-4c7a-a8a1-4fb8037ceaa0">
